web/styles/settings.css

```diff
@@ -698,6 +698,17 @@ input[type="checkbox"] {
     max-width: fit-content;
 }
 
+/* Margin on child expands the wrapper and causes tooltips
+outside the url input. We shift this margin from child to
+wrapper to remedy this. */
+.not-editable-by-user-url-input-wrapper {
+    > .settings_url_input {
+        margin-bottom: 0;
+    }
+
+    margin-bottom: 10px;
+}
```

Comment:
Maybe it would be better to refactor all existing occurrences of `.settings_url_input` such that it no longer needs a margin?

Reply:
Yeah this feels kinda hacky; @sahil839 probably has advice for what precisely to do.

Reply:
I guess we can remove the margin from `settings_url_input`, since it is anyways inconsistent with other inputs. Like `settings_text_input` also has the `margin-bottom` set as `10`, but it is overridden by other CSS which sets it to 0.
So, I guess a prep commit can be to remove this margin.

And regarding the `div`, we only really need it when we want to show a tooltip when input is disabled, and the other place where `settings_url_input`, the input element is not disabled and does not need a tooltip. And I guess removing the margin should anyways fix the tooltip without needing any extra CSS like this.

Reply:
I guess you meant `.settings_url_input` selector? It is used only in 2 places. We are removing the bottom margin for required profile fields too. Probably better to remove this as a follow-up though as it is used in org settings modal aswell.

```html
$ git grep -n ".settings_url_input"
web/styles/settings.css:741:    > .settings_url_input {
web/styles/settings.css:2114:.settings_url_input,
web/styles/settings.css:2143:    .settings_url_input,
web/templates/settings/custom_user_profile_field.hbs:29:        <input class="custom_user_field_value {{#if for_manage_user_modal}}modal_url_input{{else}}settings_url_input{{/if}}" name="{{ field.id }}" type="{{ field_type }}" value="{{ field_value.value }}" maxlength="2048" {{#unless editable_by_user}}disabled{{/unless}}/>
web/templates/settings/organization_settings_admin.hbs:146:                              name="realm_jitsi_server_url_custom_input" class="realm_jitsi_server_url_custom_input settings_url_input" maxlength="200" />
```

Reply:
Sorry, I cannot find `margin-bottom` being affected by `input[type="url"]` selector . Removing the`margin-bottom` in `.settings_url_input` completely removed this margin for me on Firefox.

Reply:
Ok. I guess I misread something. We can leave it here for now and can handle this as a follow-up.